### PR TITLE
Fix N-window selector sending mutation upon navigation to stopped workflows

### PR DESCRIPTION
**Pull request by MetRonnie**
Fixes an unreleased bug on master. Also adds some tests. Follow-up to #1755 

#### How to reproduce the bug on master

1. Start a workflow and use the UI to set the n-window to a value other than 1, e.g. 2
2. Stop the workflow
3. Navigate to the dashboard or a different workflow
4. Navigate back to the original workflow
5. You should see an error snackbar for a failed mutation

#### Check List

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Tests are included
- [x] No changelog entry needed as fixes unreleased bug

### CODE REVIEW

File: src/components/cylc/workflow/Toolbar.vue [hunk 1]
```diff
@@ -357,9 +372,20 @@ export default {
...
+      },
+      async set (val) {
+        if (val == null || this.isStopped) return
+
+        this.changingNWindow = true
+        if (await this.setGraphWindow(val)) {
+          await until(() => this.currentWorkflow?.node?.nEdgeDistance === val)
+        }
+        this.changingNWindow = false
+      },
```
Comment (MetRonnie) [inline src/components/cylc/workflow/Toolbar.vue:388]: By switching to a writeable computed property, the state relies on the subscription value and so cannot deviate from it, fixing the bug. But this means there is a delay in the update of the value displayed in the input after choosing a new value, so I added a loading spinner to give immediate feedback to the user

File: src/components/cylc/workflow/Toolbar.vue [hunk 2]
```diff
@@ -357,9 +372,20 @@ export default {
         )
       }
     },
-    nEdgeDistance () {
-      // the graph window distance reported by the scheduler
-      return this.currentWorkflow?.node?.nEdgeDistance
+    nWindow: {
+      get () {
+        // the graph window distance reported by the scheduler
+        return this.currentWorkflow?.node?.nEdgeDistance ?? 1
```
Comment (oliver-sanders) [inline src/components/cylc/workflow/Toolbar.vue:378]: For stopped workflows, the icon initially says `n=1`, but then pings to `n=0`.

Not a problem (and technically very accurate), but curious that this `n=1` default is overridden.

**Review by oliver-sanders — APPROVED** [review]

File: src/components/cylc/workflow/Toolbar.vue (hunk 2 above)
Comment (MetRonnie) [reply src/components/cylc/workflow/Toolbar.vue:378]: Not sure why, but stopped workflows have a reported `nEdgeDistance` of 0

image

File: package.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
     "preview": "node ./scripts/concurrently.cjs serve:jupyterhub preview",
     "serve": "node ./scripts/concurrently.cjs serve:jupyterhub serve:vue",
     "serve:vue": "vite --mode offline",
-    "serve:jupyterhub": "nodemon -e js,mjs,cjs,json src/services/mock/json-server.cjs",
+    "serve:jupyterhub": "nodemon src/services/mock/json-server.cjs --watch src/services/mock/",
```
Comment (markgrahamdawson) [inline package.json:20]: Whats this? is it related to the PR?

Comment (MetRonnie) [reply package.json:20]: Not related to the PR but worth getting in IMO.

The mock JSON server was unnecessarily restarting every time any JS file was edited, when it only needs to restart when files in `src/services/mock` are changed. See https://github.com/cylc/cylc-ui/pull/1786/commits/447657b1b22736945ab9879a0beaa6936b8398f3

**Review by wxtim — APPROVED** [review]
Works for me. Have not inspected the code. :)

**Review by markgrahamdawson — APPROVED** [review]

**Pull request merged**